src/net_processing.cpp

```diff
@@ -2476,19 +2476,12 @@ void PeerLogicValidation::ProcessMessage(CNode& pfrom, const std::string& msg_ty
             // We skip these operations for BLOCK_RELAY peers to avoid
             // potentially leaking information about our BLOCK_RELAY
             // connections via the addrman or address relay.
-            if (fListen && !::ChainstateActive().IsInitialBlockDownload())
+            if (fListen && !m_chainman.ActiveChainstate().IsInitialBlockDownload())
             {
-                CAddress addr = GetLocalAddress(&pfrom.addr, pfrom.GetLocalServices());
-                FastRandomContext insecure_rand;
-                if (addr.IsRoutable())
-                {
-                    LogPrint(BCLog::NET, "ProcessMessages: advertising address %s\n", addr.ToString());
-                    pfrom.PushAddress(addr, insecure_rand);
-                } else if (IsPeerAddrLocalGood(&pfrom)) {
-                    addr.SetIP(addrMe);
-                    LogPrint(BCLog::NET, "ProcessMessages: advertising address %s\n", addr.ToString());
-                    pfrom.PushAddress(addr, insecure_rand);
-                }
+                AdvertiseLocal(&pfrom);
+                const auto current_time = GetTime<std::chrono::microseconds>();
+                LOCK(pfrom.cs_sendProcessing);
```

Comment:
This mutex shouldn't actually be guarding `m_next_local_addr_send`. See https://github.com/bitcoin/bitcoin/pull/13123#issuecomment-647505130. You may as well remove that annotation if you're touching all the places where this variable is used.